Include/Acidanthera/Protocol/OcBootEntry.h

```diff
@@ -0,0 +1,140 @@
+*/
+#define LINUX_BOOT_USE_LATEST           BIT9
+/*
+  Prepend filesystem type and first 8 hex digits of PARTUUID to discovered
+  entry titles, to help in debugging where entries came from.
+*/
+#define LINUX_BOOT_ADD_DEBUG_INFO       BIT15
+
+#define LINUX_BOOT_ALL            ( \
+  LINUX_BOOT_SCAN_ESP             | \
+  LINUX_BOOT_SCAN_XBOOTLDR        | \
+  LINUX_BOOT_SCAN_EXT4            | \
+  LINUX_BOOT_SCAN_OTHER           | \
+  LINUX_BOOT_ALLOW_AUTODETECT     | \
+  LINUX_BOOT_USE_LATEST           | \
+  LINUX_BOOT_ADD_DEBUG_INFO       \
+  )
+
+/*
+  Forward declaration of OC_BOOT_ENTRY_PROTOCOL structure.
+*/
+typedef struct OC_BOOT_ENTRY_PROTOCOL_ OC_BOOT_ENTRY_PROTOCOL;
+
+/**
+  Return list of OpenCore boot entries associated with filesystem.
+
+  @param[in]  Device            The handle of the device to scan. NULL is passed in to
+                                request custom entries. All implementations must support a
+                                NULL input value, but may immediately return EFI_NOT_FOUND
+                                if they do not provide any custom entries.
+  @param[out] BootEntries       List of boot entries associated with the filesystem.
+                                On EFI_SUCCESS BootEntries must be freed by the caller
+                                with FreePool after use, and each individual boot entry
+                                should eventually be freed by FreeBootEntry (as for boot
+                                entries created within OC itself).
+                                Does not point to allocated memory on return, if any status
+                                other than EFI_SUCCESS was returned.
+  @param[out] NumEntries        The number of entries returned in the BootEntries list.
+                                If any status other than EFI_SUCCESS was returned, this
+                                value may not have been initialised and should be ignored
+                                by the caller.
+
+  @retval EFI_SUCCESS           At least one matching entry was found, and the list and
+                                count of boot entries has been returned.
+  @retval EFI_NOT_FOUND         No matching boot entries were found.
+  @retval EFI_OUT_OF_RESOURCES  Memory allocation failure.
+  @retval other                 An error returned by a sub-operation.
+**/
+typedef
+EFI_STATUS
+(EFIAPI *OC_GET_BOOT_ENTRIES) (
+  IN   EFI_HANDLE               Device,
+  OUT  OC_PICKER_ENTRY          **Entries,
+  OUT  UINTN                    *NumEntries
+  );
+
+/**
+  Free list of OpenCore boot entries from previous call to OC_GET_BOOT_ENTRIES.
+
+  @param[in]  BootEntries       List of boot entries, as returned by previous call.
+  @param[in]  NumEntries        The number of entries, as returned by previous call.
+**/
+typedef
+VOID
+(EFIAPI *OC_FREE_BOOT_ENTRIES) (
+  IN   OC_PICKER_ENTRY          **Entries,
+  IN   UINTN                    *NumEntries
```

Comment:
Why is this a pointer?

Reply:
Why is `*NumEntries` a pointer, yes? It zeros the value on call. I guess cannot enforce that all implementations of Boot Entry Protocol do that nicely... Equally `**Entries` is pointer to pointer in order to zero that too. Overkill/bad idea?

Reply:
I have not seen that before, not even in other codebases. For `NumEntries` I think it's unintuitive, it's a bad example, but what if you want to log how many entries you just destructed? It may make sense to preserve it. I makes no sense to preserve the pointer, but also it's unconventional I guess?

Reply:
There's definitely a few places around OC where **ptr is passed in for freeing pointer memory - so leave **ptr but just change to NumEntries? Or change both? Definitely can do - perhaps makes more sense not to use ** at something which is more of an interface than some method internal to OC.

Reply:
Updated in latest push to PR